hyperspy/_signals/complex_signal.py

```diff
@@ -260,6 +261,59 @@ def angle(self, deg=False):
         return super().angle(angle, deg=deg)
     angle.__doc__ = ComplexSignal_mixin.angle.__doc__
 
+    def argand_diagram(self, size=[256, 256], display_range=None):
+        """
+        Calculate and plot Argand diagram of complex signal
+
+        Parameters
+        ----------
+        size : [int, int], optional
+            Size of the Argand plot in pixels
+            (Default: [256, 256])
+        display_range : array_like, shape(2,2) or shape(2,) optional
+            The position of the edges of the diagram
+            (if not specified explicitly in the bins parameters): [[xmin, xmax], [ymin, ymax]].
+            All values outside of this range will be considered outliers and not tallied in the histogram.
+            (Default: None)
+
+        Returns
+        -------
+        sap:
+            Argand diagram plot as Signal2D
+
+        Examples
+        --------
+        >>> import hyperspy.api as hs
+        >>> holo = hs.datasets.example_signals.object_hologram()
+        >>> ref = hs.datasets.example_signals.reference_hologram()
+        >>> w = holo.reconstruct_phase(ref)
+        >>> w.argand_diagram(display_range=[-3, 3]).plot()
+
+        """
+        im = self.imag.data.ravel()
+        re = self.real.data.ravel()
+
+        if display_range:
+            if np.asarray(display_range).shape == (2, ):
+                display_range = [[display_range[0], display_range[1]],
+                                 [display_range[0], display_range[1]]]
+            elif np.asarray(display_range).shape != (2, 2):
+                raise ValueError('display_range should be array_like, shape(2,2) or shape(2,).')
+
+        ap = np.histogram2d(re, im, bins=size, range=display_range)
+        sap = Signal2D(ap[0].T)
+        sap.metadata = self.metadata.deepcopy()
+        sap.metadata.General.title = 'Argand diagram of {}'.format(self.metadata.General.title)
+        sap.metadata.Signal.signal_type = 'signal2d'
+        sap.axes_manager.signal_axes[0].name = 'Real'
+        sap.axes_manager.signal_axes[0].offset = ap[1][0]
```

Comment:
If you take into account the change suggested above, then change ap[1] here and in the next line to xedges

Reply:
Set the units too when they are available?

Reply:
@ericpre With units, I'm not sure. These are just _Im_ and _Re_ parts of complex numbers which are kind of unitless by definition. On the other hand one may think of applying the same units as the quantity of original signal, though I'm not sure if that is correct neither. Moreover I can't remember any complex value that would also have units as having an imaginary part would mean that it is something that doesn't exist physically. Correct me if I'm wrong. 
Regardless your answer, I will try to ask smart people around me to get better understanding.

Reply:
I don't know/can't think of a case where it would be used but I would imagine that it would exist!
In any case, from a purely generic point of view, we could extract the units from `s.real.metadata.Signal.quantity` and `s.imag.metadata.Signal.quantity`, since it should in bracket of this metadata.
Similarly, the name of the axis could be pull from there, when available.

Reply:
Sounds good! I had a thought about complex number in physics and the most obvious one is the electrical impedance which have units of Ohm. In this case the units for each of Re and Im parts are also Ohms, which is logical since the complex number is a sum of these two. So your suggestion should do.

Reply:
@ericpre It appeared to be more complex (since we are dealing with complex numbers))) than I thought. The issue is the absence of a standard for using `quantity` attribute. I approached it voluntarily by parsing the `quantity` if it is available which isn't a robust solution against non-trivial use cases.  Please have a look and let me know if it makes sense or if you would have better suggestions. I'll add corresponding test once it is decided how this part will look like.

Reply:
e.g.
```python
>>> from hyperspy.misc.utils import parse_quantity
>>> parse_quantity('Resistance (Ohm)')
('Resistance ', 'Ohm')

Reply:
I open a PR (https://github.com/woozey/hyperspy/pull/35) to this branch to tweak the parsing of the quantity metadata and I think the approach should be good now. Please let me know what you think.

Reply:
@ericpre Added test for units assignment. Should be ready now